sql/src/main/java/io/crate/operation/reference/sys/node/DiscoveryNodeContextFieldResolver.java

```diff
@@ -0,0 +1,251 @@
+            return context;
+        }
+
+        List<ColumnIdent> processedColIdents = new ArrayList<>();
+        for (ReferenceIdent ident : referenceIdents) {
+            ColumnIdent columnIdent = ident.columnReferenceIdent().columnIdent();
+            if (SysNodesTableInfo.IDENT.equals(ident.tableIdent()) && !processedColIdents.contains(columnIdent)) {
+                Consumer<DiscoveryNodeContext> consumer = columnIdentToContext.get(columnIdent);
+                if (consumer == null) {
+                    throw new IllegalArgumentException(
+                        String.format("Cannot resolve DiscoveryNodeContext field for \"%s\" column ident.", columnIdent)
+                    );
+                }
+                consumer.accept(context);
+                processedColIdents.add(columnIdent);
+            }
+        }
+        return context;
+    }
+
+    private final Map<ColumnIdent, Consumer<DiscoveryNodeContext>> columnIdentToContext =
+        ImmutableMap.<ColumnIdent, Consumer<DiscoveryNodeContext>>builder()
+            .put(SysNodesTableInfo.Columns.ID, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.id(BytesRefs.toBytesRef(clusterService.localNode().id()));
+                }
+            })
+            .put(SysNodesTableInfo.Columns.NAME, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.name(BytesRefs.toBytesRef(clusterService.localNode().name()));
+                }
+            })
+            .put(SysNodesTableInfo.Columns.HOSTNAME, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    try {
+                        context.hostname(BytesRefs.toBytesRef(InetAddress.getLocalHost().getHostName()));
+                    } catch (UnknownHostException e) {
+                        // What should we do?
+                    }
+                }
+            })
+            .put(SysNodesTableInfo.Columns.REST_URL, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    if (clusterService.localNode() != null) {
+                        String url = clusterService.localNode().attributes().get("http_address");
+                        context.restUrl(BytesRefs.toBytesRef(url));
+                    }
+                }
+            })
+            .put(SysNodesTableInfo.Columns.PORT, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    Integer http = null;
+                    Integer transport;
+
+                    if (nodeService.info().getHttp() != null) {
```

Comment:
`info()` is creating new objects when called. Please use local variable to avoid doing that twice